base/shell/explorer/settings.cpp

```diff
@@ -32,7 +32,6 @@ BOOL TaskbarSettings::Save()
     SHSetValueW(hkExplorer, L"Advanced", L"TaskbarSizeMove", REG_DWORD, &bAllowSizeMove, sizeof(bAllowSizeMove));
     sr.cbSize = sizeof(sr);
     SHSetValueW(hkExplorer, L"Advanced", L"TaskbarSmallIcons", REG_DWORD, &bSmallIcons, sizeof(bSmallIcons));
-    SHSetValueW(hkExplorer, L"Advanced", L"CompactTrayIcons", REG_DWORD, &bCompactTrayIcons, sizeof(bCompactTrayIcons));
```

Comment:
Why is this save not needed?

Reply:
The save function is called only when applying taskbar settings from the Taskbar Properties window. Since there is no plan to expose this functionality through the UI, keeping this here is a bug.

For example, if you run ROS explorer on Windows Server 2003 and change any setting, you will see that it creates the registry values `PerferDateOverWeekday`, `TaskbarSd`, and `CompactTrayIcons` in `HKCU\...\Explorer\Advanced` even though these settings were not changed. Also, if you change the registry value yourself, then change any setting in Taskbar Properties, it resets this value to the default value unless you reload explorer before changing the taskbar setting. This behavior is a bug.

In a separate PR, I would like to make a distinction between settings exposed by the properties UI and settings that are not. The settings not exposed via the properties window should not be saved when calling `Save()`, since they can only be set manually by the user.

Reply:
> since they can only be set manually

So this requires the user to "know" about these extra ROS settings?

Reply:
Correct. The expectation is that most people will not want to change it and that adding it to the taskbar properties window would not be worth the additional complexity.